dotnet/src/SemanticKernel.Abstractions/Services/OrderedAIServiceSelector.cs

```diff
@@ -107,4 +118,12 @@ kernel.Services is IKeyedServiceProvider ?
 
         return null;
     }
+
+    /// <inheritdoc/>
+    public bool TrySelectAIService<T>(Kernel kernel, KernelFunction function, KernelArguments arguments, [NotNullWhen(true)] out T? service, out PromptExecutionSettings? serviceSettings) where T : class, IAIService
+        => this.TrySelect(kernel, function, arguments, out service, out serviceSettings);
+
+    /// <inheritdoc/>
+    public bool TrySelectChatClient<T>(Kernel kernel, KernelFunction function, KernelArguments arguments, [NotNullWhen(true)] out T? service, out PromptExecutionSettings? serviceSettings) where T : class, IChatClient
```

Comment:
Where do you end up needing this? Is there an equivalent TrySelectChatCompletionService somewhere?

Reply:
Yes, `TrySelectAIService`, `OrderedServiceSelector` implements both `IAIService` and `IChatClientSelectors`, this method allow it to be smart at the IChatClient lookups.